mlir/lib/Analysis/PresburgerSet.cpp

```diff
@@ -318,6 +318,52 @@ bool PresburgerSet::findIntegerSample(SmallVectorImpl<int64_t> &sample) {
   return false;
 }
 
+PresburgerSet PresburgerSet::coalesce() {
+  assert(nSym == 0 && "symbolic sets are not yet supported by coalesce");
+  PresburgerSet newSet =
+      PresburgerSet::getEmptySet(this->getNumDims(), this->getNumSyms());
+  ArrayRef<FlatAffineConstraints> basicSetVector =
```

Comment:
I don't understand why this is needed. Just use `flatAffineConstraints` directly?